shared-services/src/files/cloudinary-file.service.ts

```diff
@@ -0,0 +1,21 @@
+import { SetStateAction } from "react"
```

Comment:
Why do we have a dependency on react?  Would we be able to call setProgressValue through the Promise instead to avoid introducing additional dependencies?

Reply:
Digging deeper, it looks like that callback ultimately just gets called by an `onUploadProgress` event handler in axios.  Can we make it more generic here to have a function that just takes a number from 0-100 representing progress, then have a wrapper calling the react-specific function in `cloudinaryFileUploader`?

Reply:
Progress value affects the UI since there is a progress attribute of the [dropzone](https://github.com/bloom-housing/ui-components/blob/main/src/forms/Dropzone.tsx#L50) which renders the progress bar while the upload is happening. All the (current) callers of this method rely on the progress value being updated by cloudinaryFileUploader while the upload happens using the function that updates the state. If I understand correctly, what you suggested would still update the new number you define, but it's not tied back to the progressValue that the dropzone uses [here](https://github.com/bloom-housing/bloom/blob/main/sites/partners/src/components/listings/PaperListingForm/sections/ListingPhoto.tsx#L231).

Reply:
Let's try to limit dependencies where we can, otherwise we run the risk of both overcomplicating our API and introducing unneeded dependencies elsewhere (ie possibly having to add react to the backend in the future).  This package/class doesn't need react itself, so we don't need to include it.  Where calling code may already be using react, we can use a thin translation to abstract things away.  In my example above, the `putFile` method would only need a callback function that accepts a single float/int.  The calling code could easily provide a function with that signature which just does a type conversion before calling the react-specific function that already exists.  It would be maybe one extra line of code.  That would be preferable IMO

Reply:
Digging a little bit deeper, it looks like the signature for the progress callback on the [CloudinaryUpload](https://github.com/bloom-housing/ui-components/blob/abfe88642a27d79b4a24a0d65c2ab312057ad46e/src/forms/CloudinaryUpload.ts#L5) function also only uses a number type.  In places where [useState](https://github.com/bloom-housing/bloom/blob/815da6e69b2f212547ab52fb4540a712271d4945/sites/partners/src/components/listings/PaperListingForm/sections/LotteryResults.tsx#L35) or [setProgressValue](https://github.com/bloom-housing/bloom/blob/815da6e69b2f212547ab52fb4540a712271d4945/sites/partners/src/components/listings/PaperListingForm/sections/LotteryResults.tsx#L50) are called, it's often with an int literal.

Reply:
Done, sorry it took me a bit to understand and thanks for the explanation. I spun up the site and tested uploading a new listing photo to the partners site and it is still working well.